articles/creating-a-hangman-game-with-vanilla-javascript/index.md

```diff
@@ -0,0 +1,238 @@
+
+Now let’s actually guess the letters
+###handleGuess() function
+Firstly, we are going to pass down a `chosenLetter`  . This function handles the letters to be guessed and determines if the answer is true or false. In the function, we are going to say if the chosen letter does not exist, we want to go ahead and push `chosenLetter` into the array but if it does exist, then we would do nothing.  Next, you run an `if` statement saying `if (answer.indexOf(chosenLetter) >= 0)` meaning that if it exists, then run the `guessed();` function so that it updates the letters. Now we have to start incrementing the number of wrong guesses if the player gets it wrong.  We can do that by using the `elseif` statement inside this function and passing in the “answer.indexOf(chosenLetter) === -1” then mistakes is going to be added by one. Here’s  what the code will look like:
+```javascript
+function handleGuess(chosenLetter) {
+  guessed.indexOf(chosenLetter) === -1 ? guessed.push(chosenLetter) : null;
+  document.getElementById(chosenLetter).setAttribute('disabled', true);
+
+  if (answer.indexOf(chosenLetter) >= 0) {
+    guessedWord();
+  } else if (answer.indexOf(chosenLetter) === -1) {
+    mistakes++;
+    updateMistakes();
+  }
+}
+```
+
+###updateMistakes() function
+You also have to run the updateMistakes(); function so that it actually updates the number since we are not using any framework like ReactJS or the rest of them.  Here’s how it would look like in the code:
+```javascript
+function updateMistakes() {
+  document.getElementById('mistakes').innerHTML = mistakes;
+}
+```
+Now, the number of wrong guesses will always keep updating by +1.
+The next thing to do now is to check for wins and losses so that the player does not just keep playing endlessly. To do that you run a checkifGameWon(); in the `if` statement of the `handleGuess();` function and a `checkIfGameLost();` in the corresponding `elseif` statement.
+###checkIfGameWon() function
+We are going to start by creating an if statement saying if “wordStatus” is equal to the answer, the keyboard will print out “You won!!!”. Here is the code below:
+```javascript
+function checkIfGameWon() {
+  if (wordStatus === answer) {
+    document.getElementById('keyboard').innerHTML = 'You Won!!!';
+  }
+}
+```
+
+###checkIfGameLost() function
+Here’s the code to check if the game lost.
+```javascript
+function checkIfGameLost() {
+  if (mistakes === maxWrong) {
+    document.getElementById('wordSpotlight').innerHTML = 'The answer was: ' + answer;
+    document.getElementById('keyboard').innerHTML = 'You Lost!!!';
+  }
+}
+```
+In the code above, We set the game to print out “You lost!!!” when the player has reached the maximum number of wrong guesses. We also set the game to print out the correct answer too.
+reset() function
+In this function, we are going to set everything back to default.  Here is how it looks in the code:
```

Comment:
I guess the line `reset() function` is a heading? Please use ### to format it to be a heading.